packages/dev/core/src/Rendering/boundingBoxRenderer.ts

```diff
@@ -483,6 +533,229 @@ export class BoundingBoxRenderer implements ISceneComponent {
+     * calling engine.drawElementsType for each bounding box in renderList,
+     * making every bounding box 1 or 2 draw call.
+     * When enabled, it collects bounding boxes to render,
+     * and render all boxes in 1 or 2 draw call.
+     * This could make the rendering with many bounding boxes much faster than not enabled,
+     * but could result in a difference in rendering result if
+     * {@link BoundingBoxRenderer.showBackLines} enabled,
+     * because drawing the black/white part of each box one after the other
+     * can be different from drawing the black part of all boxes and then the white part.
+     * Also, when enabled, events of {@link BoundingBoxRenderer.onBeforeBoxRenderingObservable}
+     * and {@link BoundingBoxRenderer.onAfterBoxRenderingObservable} would only be triggered once
+     * for one rendering, instead of once every bounding box.
+     * Events would be triggered with a dummy box to keep backwards compatibility,
+     * the passed bounding box has no meaning and should be ignored.
+     * @param val whether to use instanced rendering
+     */
+    set useInstances(val: boolean) {
+        this._useInstances = val;
+        if (this._colorShader) {
+            this._colorShader.setDefine("INSTANCES", val);
+        }
+        if (!val) {
+            this._cleanupInstances();
+        }
+    }
+
+    get useInstances(): boolean {
+        return this._useInstances;
+    }
+
+    /**
+     * Instanced render the bounding boxes of a specific rendering group
+     * @internal
+     * @param renderingGroupId defines the rendering group to render
+     */
+    public _renderInstanced(renderingGroupId: number): void {
+        if (this.renderList.length === 0 || !this.enabled) {
+            return;
+        }
+        this._prepareResources();
+
+        if (!this._colorShader.isReady()) {
+            return;
+        }
+
+        const colorShader = this._colorShader;
+        let matrices = this._matrices;
+        const expectedLength = this.renderList.length * 16;
+        if (!matrices || matrices.length < expectedLength || matrices.length > expectedLength * 2) {
+            matrices = new Float32Array(expectedLength);
+            this._matrices = matrices;
+        }
+
+        this.onBeforeBoxRenderingObservable.notifyObservers(dummyBoundingBox);
+
+        let instancesCount = 0;
+
+        for (let boundingBoxIndex = 0; boundingBoxIndex < this.renderList.length; boundingBoxIndex++) {
+            // p = performance.now()
+            const boundingBox = this.renderList.data[boundingBoxIndex];
+            if (boundingBox._tag !== renderingGroupId) {
+                continue;
+            }
+
+            const min = boundingBox.minimum;
+            const max = boundingBox.maximum;
+
+            const diff = max.subtractToRef(min, vec2);
+            const median = min.addToRef(diff.scaleToRef(0.5, vec), vec);
+
+            const m = (mat1 as any)._m;

```

Comment:
cc @BabylonJS/core-team-microsoft 

I think we should make `Matrix._m` `public` / `@internal`, to allow such use cases.

It doesn't seem right to cast as `any` to get `_m`, it might break if we change the implementation of `Matrix` in the future. But at the same time, such use cases where we know what we're doing and don't need to update the `updateFlag` should also be possible.

Reply:
Yes, but it is readonly / immutable, you can't do `matrix.m[x] = y`.

Reply:
not opposed to put it public internal as I do not want it as full public considering the error risk :-)

Reply:
Since [`Matrix.asArray`](https://github.com/BabylonJS/Babylon.js/blob/fe7ee12c805f26b89b9bf1c753f3a45e8b20e833/packages/dev/core/src/Maths/math.vector.ts#L6252) returns the internal array directly, it can be used in this case without force cast.